lib/matplotlib/lines.py

```diff
@@ -158,8 +158,12 @@ def _slice_or_none(in_v, slc):
                 raise ValueError(
                     "markevery is specified relative to the axes size, but "
                     "the line does not have a Axes as parent")
+
             # calc cumulative distance along path (in display coords):
-            disp_coords = affine.transform(tpath.vertices)
+            fin = np.isfinite(verts).all(axis=1)
+            fverts = verts[fin]
+            disp_coords = affine.transform(fverts)
```

Comment:
This seems the easiest, and has the feature of only testing shown markers for visual distance.  Not always what folks want, but at least doesn't error, and has the benefit of simplicity.

Reply:
Are you ok with this @rc or did you want to argue for a different algorithm?   @oscargus this changed since your approval.  Still OK?

Reply:
Still OK! I think we have similar views: no error thrown and some output that makes sense (although it is and always will be possible to find cases that gives "sub-optimal" results).

And nicer with a shorter/simpler solution as well.

Reply:
Yes, this is OK!

Reply:
@rc can you squash to one commit, and then I will merge...  If you have not done a rebase before, make sure to make a backup branch so you can get back to this point.  But basically

```
git fetch upstream
git rebase -i upstream/master
```
and choose squash for all your commits except the first one.

Reply:
Done.